### ISSUE TITLE
Screenshot with SDL1 doesn't reflect original colors

### ISSUE THREAD

**Issue by hihihirokane** [joncampbell123/dosbox-x#727]
This is a screenshot (F12+Shift+S) by the BETA (2018-5-18) version built with in-tree SDL1.x on OS X Mountain Lion. White color turns into yellow, green and red interchange their colors, and so on. 
![dosbox_004](https://user-images.githubusercontent.com/1503377/40281643-f006635c-5c53-11e8-9b39-a15a9a3b8998.png)
As a reference, a binary built with SDL2 didn't change colors on the screenshot. This may have nothing to do with PC-98 mode because 'machine=svga_s3' also cause this.
![dosbox_005](https://user-images.githubusercontent.com/1503377/40281646-f5277b5a-5c53-11e8-92d6-014b5141d21d.png)

**Comment by emendelson** [comment]
It may be unlikely, but could this problem have anything to do with the various macOS problems in this thread:

https://www.vogons.org/viewtopic.php?f=32&t=54314

**Comment by joncampbell123** [comment]
Right. Mac OS X bitmaps follow a strange RGBA pattern where the alpha channel is in the low 8 bits. I thought I fixed that...

**Comment by emendelson** [comment]
I think it's fixed in the way the macOS displays the screen - the reported problem occurs only screenshots, it seems.

**Comment by joncampbell123** [comment]
Ah, so the C_SHOT code needs to reverse RGBA again for libpng.

Does this also happen with video capture?

**Comment by emendelson** [comment]
I hope the OP will report on that - I don't have a current Mac version running, but can build one later on if needed.

**Comment by emendelson** [comment]
OK, I built the latest code, and can't find any way to make a screen shot or record video. The OP says that F12+Shift+S creates a screen shot, but that seems only to enter "S" at the command prompt.

I used the "build" script to build. Should I try something else?

**Comment by hihihirokane** [comment]
I may have configured the mapper myself about screenshot and video. I shouldn't have posted confusing description.

**Comment by emendelson** [comment]
OK - I'll stay out of this from here on...!

**Comment by joncampbell123** [comment]
@emendelson The screenshot function requires that libz (zlib) and libpng are available to compile against. Are the headers and libraries available on your system when you build?

**Comment by emendelson** [comment]
OK, I used to use MacPorts, but stopped because I kept making apps that wouldn't work on other people's machines (long story). But it would be terrific if you could integrate zlib and libpng into the macOS version. (The only GitHub project of mine that actually works, or used to work, was a script called OneStopDOSBoxOSX that downloads the libraries and runs a build script that uses then; most of that script was based on ideas thought up by Dominus at Vogons.org, but it made it possible to build SVN in macOS with a few clicks.)

**Comment by joncampbell123** [comment]
It's possible your application linked to the libraries dynamically.

I've learned long ago that on Windows and Mac OS X it's generally easier to either statically link the library into your program or include the DLL or dylib files in your application. Never assume the other systems have it. You make your program more portable that way.

**Comment by joncampbell123** [comment]
Linux is generally an exception because people are more likely to compile from source than run a binary.

**Comment by emendelson** [comment]
That's exactly what I've learned to do - after a lot of frustrations figuring out how to link statically. Getting that to work with DOSBox SVN on the Mac was a nightmare, and I still don't know how to do it - I simply follow Dominus's method without understanding it, and it works.

**Comment by scpedicini** [comment]
@joncampbell123 Was this issue ever resolved when recording video/screenshots?

I'm seeing it happening in `v2024.07.01` of Dosbox-X.

![image](https://github.com/user-attachments/assets/07c09ff8-da5b-4d37-ab0d-76fb9b7c4a20)
![image](https://github.com/user-attachments/assets/a051d2b6-4f85-4db5-b13a-f87cb9812163)
